### PR TITLE
[DT-142] ih-pre-commit integration

### PR DESCRIPTION
**Pull request by SteveRuble**
This adds a new, custom pre-commit handler (ih-pre-commit) which resides in the image-builder repository. This change updates the git step to remove its installation of the standard pre-commit hook, and updates the toolrepos step to install the ih-pre-commit step as a template and to update all existing IH repositories. 

This also introduces a shell script which can be run to pull the latest HEAD of a repo, or clone the repo if it is not present. This is used by the pre-commit hook to keep itself up-to-date.

This also adds a couple useful tools as dependency, yq and ripgrep.

### CODE REVIEW

**Comment by SteveRuble** [comment]
We don't want to merge this until https://github.com/ConsultingMD/image-builder/pull/87 is merged

**Review by francisco-hurtado — APPROVED** [review]

File: lib/core/git/step.sh [hunk 1]
```diff
@@ -53,17 +53,6 @@ function ih::setup::core.git::install() {
   # make git use ssh for everything
   git config --global url.git@github.com:ConsultingMD/.insteadOf https://github.com/ConsultingMD/
 
-  # set up pre-commit to automatically be set up for all cloned repos,
```
Comment (SteveRuble) [inline lib/core/git/step.sh:56]: Moved to `ih-pre-commit install` which is now invoked in `toolrepos`.

File: lib/core/toolrepos/step.sh [hunk 2]
```diff
@@ -37,15 +35,15 @@ function ih::setup::core.toolrepos::test-or-install() {
       return 1
     fi
     ih::log::info "Cloning engineering repo..."
-    git clone git@github.com:ConsultingMD/engineering.git --filter=blob:limit=1m --depth=5 "${GR_HOME}/engineering" || return
```
Comment (SteveRuble) [inline lib/core/toolrepos/step.sh:40]: This repo has been truncated and is no longer GB in size, so we can just clone the whole thing.

File: lib/core/toolrepos/step.sh [hunk 3]
```diff
@@ -37,15 +35,15 @@ function ih::setup::core.toolrepos::test-or-install() {
...
+    git clone git@github.com:ConsultingMD/engineering.git --filter=blob:limit=1m "${GR_HOME}/engineering" || return
   fi
 
   if [ ! -d "${GR_HOME}/image-builder" ]; then
     if [ "$1" == "test" ]; then
       return 1
     fi
     ih::log::info "Cloning image-builder repo.."
-    git clone git@github.com:ConsultingMD/image-builder.git --filter=blob:limit=1m --depth=5 "${GR_HOME}/image-builder" || return
```
Comment (SteveRuble) [inline lib/core/toolrepos/step.sh:48]: `image-builder` was never that big, so we can just clone the whole thing.

File: lib/core/toolrepos/step.sh [hunk 4]
```diff
@@ -67,14 +65,37 @@ function ih::setup::core.toolrepos::test-or-install() {
...
-  export IH_WANT_RE_SOURCE=1
+  PRE_COMMIT_HOOK_DST="${IH_DIR}/git-template/hooks/pre-commit"
+  if [ ! -f "$PRE_COMMIT_HOOK_DST" ]; then
+    if [ "$1" == "test" ]; then
+      return 1
+    fi
 
-  cp -f "$toolsrepo_src_path" "$toolsrepo_tgt_path"
+    export IMAGE_BUILDER_ROOT="${IH_HOME}/image-builder"
+    (
```
Comment (SteveRuble) [inline lib/core/toolrepos/step.sh:83]: Create a subshell because we're `cd`-ing over to `image-builder`.

File: lib/core/toolrepos/step.sh [hunk 5]
```diff
@@ -84,11 +105,12 @@ function ih::setup::core.toolrepos::set-auto-update-repositories-job() {
   PLIST_FILE="com.includedhealth.auto-update-repositories"
   LAUNCH_AGENTS_PATH="${HOME}/Library/LaunchAgents/${PLIST_FILE}.plist"
 
+  # shellcheck disable=SC2001
```
Comment (SteveRuble) [inline lib/core/toolrepos/step.sh:108]: The updated `shellcheck` linter has a suggestion here but I don't understand if it should be accepted:
```
  GR_HOME_ESC=$(echo "$GR_HOME" | sed 's_/_\\/_g')
                ^-- SC2001 (style): See if you can use ${variable//search/replace} instead.

For more information:
  https://www.shellcheck.net/wiki/SC2001 -- See if you can use ${variable//se...

```

**Review by francisco-hurtado — APPROVED** [review]

**Pull request merged**